python_modules/dagster/dagster/core/execution/plan/plan.py

```diff
@@ -758,13 +758,36 @@ def build_subset_plan(
         step_output_versions = check.opt_dict_param(
             step_output_versions, "step_output_versions", key_type=StepOutputHandle, value_type=str
         )
-        step_handles_to_execute = [StepHandle.parse_from_key(key) for key in step_keys_to_execute]
 
+        step_handles_to_execute_set: Set[StepHandleUnion] = set()
         bad_keys = []
-        for handle in step_handles_to_execute:
+        for key in step_keys_to_execute:
+            handle = StepHandle.parse_from_key(key)
+
             if handle not in self.step_dict:
+                if isinstance(handle, ResolvedFromDynamicStepHandle):
+                    unresolved_handle = handle.unresolved_form
+                    if unresolved_handle in self.step_dict and unresolved_handle in [
+                        StepHandle.parse_from_key(key) for key in step_keys_to_execute
+                    ]:
```

Comment:
should structure this so were not re-parsing N keys every iteration

Reply:
refactored